File: 0389-find-the-difference/0389-find-the-difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        // use bitwise XOR on all letters in both s and t to reveal the remaining letter
        
        char x = '\0';
        
        for (char& c: s) {
            x ^= c;
        }
        for (char& c: t) {
            x ^= c;
        }
        
        return x;
    }
};

File: 0389-find-the-difference/0389-find-the-difference.rs
impl Solution {
    pub fn find_the_difference(s: String, t: String) -> char {
        // use bitwise XOR on all letters in both s and t to reveal the remaining letter
        
        let mut x: u8 = 0;
        
        // perform bitwise XOR in u8 (which is bitwise equivalent to char)
        for c in s.chars() {
            x ^= (c as u8);
        }
        for c in t.chars() {
            x ^= (c as u8);
        }
        
        return x as char;
    }
}